ddl/ddl_api.go

```diff
@@ -2628,6 +2628,12 @@ func (d *ddl) createTableWithInfoPost(
 			return errors.Trace(err)
 		}
 	}
+	// For issue https://github.com/pingcap/tidb/issues/46093
+	if tbInfo.AutoIncIDExtra != nil && *tbInfo.AutoIncIDExtra != 0 {
```

Comment:
given the `*tbInfo.AutoIncIDExtra != 0` condition i think making `AutoIncIDExtra` an `int64` rather than `*int64` is good enough.

we could add a `omitempty` to prevent it from showing up in JSON on zero

Reply:
Good suggestion, I'll take a try.